File: C++/3list/Circle.cpp
/*
@author: Stanisław Woźniak
@title: circle.hpp
*/
#include "Circle.hpp"
#include <cmath>


Circle::Circle(double radius){
  this -> radius = radius;
}

double Circle::field(){
  return radius*radius*M_PI;
}

double Circle::perimeter(){
  return radius*2*M_PI;
}

File: java/3list/src/com/stanislaw/moreFigures/Circle.java
package com.stanislaw.moreFigures;
/*
@author: Stanisław Woźniak
@version: 1.0
*/
import com.stanislaw.core.FigureImpl;

public class Circle extends FigureImpl {

    private double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    @Override
    public double field() {
        return this.radius*this.radius*Math.PI;
    }

    @Override
    public double perimeter() {
        return 2*Math.PI*this.radius;
    }
}
